myhomework3.php

```diff
@@ -0,0 +1,141 @@
+<?php
+function readHttpLikeInput()
+{
+    $f = fopen('php://stdin', 'r');
+    $store = "";
+    $toread = 0;
+    while ($line = fgets($f)) {
+        $store .= preg_replace("/\r/", "", $line);
+        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+            $toread = $m[1] * 1;
+        if ($line == "\r\n")
+            break;
+    }
+    if ($toread > 0)
+        $store .= fread($f, $toread);
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body)
+{
+    $response = "HTTP/1.1 $statuscode $statusmessage\n";
+    $response .= "Date: " . date("l, d F  Y h:i:sa") . "\n";
+    $response .= "Server: Apache/2.2.14 (Win32)\n";
+    $response .= "Content-Length: " . strlen($body) . "\n";
+    $response .= "Connection: Closed\n";
+    $response .= "Content-Type: text/html; charset=utf-8\n";
+    $response .= "\n$body";
+
+    echo $response;
+}
+
+function processHttpRequest($method, $uri, $headers, $body)
+{
+    $statuscode = getStatusCode($method, $uri);
```

Comment:
that's very bad to do like this. You have lots of checks of the same conditions.
read about validation and review this logic. 
your goal is to make your code simpler, but not more complicated

Reply:
will review this task further after needed changes

Reply:
actually, will review all the rest tasks, after code updates. next task will have same comment about over complicated logic